Ignore certain languages from being parsed by Prism

There are certain gitbook plugins that uses the three back-ticks to denote the code to be rendered.
See [mermaid-full usage](https://github.com/yanni4night/gitbook-plugin-mermaid-full/issues/3)
And prism gives a warning like:
```
Failed to load prism syntax: graph
{ Error: Cannot find module 'prismjs/components/prism-graph.js' ...
```

It would be nice to have an option to ignore certain "languages" from being parsed by prism.

--- a/README.md
+++ b/README.md
@@ -48,6 +48,20 @@ Support non-standard syntax prefixes by aliasing existing prefixes.
 }
 ```
 
+### `ignore`
+Due to other plugins using code block notion to denote other functionality, you can ignore certain langs
+
+```json
+"pluginsConfig": {
+  "prism": {
+    "ignore": [
+      "mermaid", //used by https://github.com/yanni4night/gitbook-plugin-mermaid-full
+      "eval-js" //used by https://github.com/brian-dawn/gitbook-plugin-klipse
+    ]
+  }
+}
+```
+
 ### Prism Themes
 
 [https://github.com/PrismJS/prism](https://github.com/PrismJS/)

--- a/README.md
+++ b/README.md
@@ -55,8 +55,8 @@ Due to other plugins using code block notion to denote other functionality, you 
 "pluginsConfig": {
   "prism": {
     "ignore": [
-      "mermaid", //used by https://github.com/yanni4night/gitbook-plugin-mermaid-full
-      "eval-js" //used by https://github.com/brian-dawn/gitbook-plugin-klipse
+      "mermaid",
+      "eval-js"
     ]
   }
 }

--- a/index.js
+++ b/index.js
@@ -86,10 +86,16 @@ module.exports = {
 
       var highlighted = '';
       var userDefined = getConfig(this, 'pluginsConfig.prism.lang', {});
+      var userIgnored = getConfig(this, 'pluginsConfig.prism.ignore', []);
 
       // Normalize language id
       var lang = block.kwargs.language || DEFAULT_LANGUAGE;
       lang = userDefined[lang] || MAP_LANGUAGES[lang] || lang;
+
+      // Check to see if the lang is ignored
+      if (userIgnored.indexOf(lang) === -1) {
+        return block.body
+      }
 
       // Try and find the language definition in components folder
       if (!languages[lang]) {

--- a/index.js
+++ b/index.js
@@ -93,7 +93,7 @@ module.exports = {
       lang = userDefined[lang] || MAP_LANGUAGES[lang] || lang;
 
       // Check to see if the lang is ignored
-      if (userIgnored.indexOf(lang) === -1) {
+      if (userIgnored.indexOf(lang) > -1) {
         return block.body
       }
 

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "gitbook-plugin-prism",
   "description": "Prism highlighting for gitbook",
   "main": "index.js",
-  "version": "2.2.1",
+  "version": "2.3.1",
   "scripts": {
     "lint": "eslint index.js test.js",
     "test": "npm run lint && tape test.js"

